harness/determined/pytorch/dsat/_dsat_search_method.py

```diff
@@ -117,34 +123,38 @@ def mbs_in_lineage(self) -> Set["DSATTrial"]:
 
     @property
     def stage(self) -> int:
-        return self.ds_config.get("zero_optimization", {}).get("stage", 0)
+        return int(self.ds_config.get("zero_optimization", {}).get("stage", 0))
 
     @property
-    def fp16(self) -> int:
-        return self.ds_config.get("fp16", {}).get("enabled") or False
+    def fp16(self) -> bool:
+        return bool(self.ds_config.get("fp16", {}).get("enabled")) or False
 
     @property
     def mbs(self) -> int:
-        return self.ds_config["train_micro_batch_size_per_gpu"]
+        return self.ds_config.get("train_micro_batch_size_per_gpu", 1)
```

Comment:
A little concerned that this could cause confusing errors in cases where `"train_micro_batch_size_per_gpu"` is omitted from the `ds_config`. Basically, `"train_micro_batch_size_per_gpu"` should always be a key and if its not, there's probably and error *and* the proper value of `"train_micro_batch_size_per_gpu"` is probably not 1.

Reply:
I will add asserts instead in a second pass